pkg/cloud/scope/rosacontrolplane.go

```diff
@@ -201,3 +202,18 @@ func (s *ROSAControlPlaneScope) PatchObject() error {
 func (s *ROSAControlPlaneScope) Close() error {
 	return s.PatchObject()
 }
+
+// GetMachinePools belong to the controlPlane managed cluster.
+func (s *ROSAControlPlaneScope) GetMachinePools() ([]expclusterv1.MachinePool, error) {
+	machinePoolList := expclusterv1.MachinePoolList{}
+	listOptions := []client.ListOption{
+		client.InNamespace(s.Cluster.Namespace),
+		client.MatchingLabels(map[string]string{clusterv1.ClusterNameLabel: s.Cluster.Name}),
+	}
+
+	if err := s.Client.List(context.TODO(), &machinePoolList, listOptions...); err != nil {
```

Comment:
please pass a `context` parament to this function and use it here instead of `context.TODO()`